src/walk_dirs.rs

```diff
@@ -159,16 +160,26 @@ pub fn discover_fences_and_files(
 
                                     let source_file_path = slashed_as_relative_path(&file_path);
 
-                                    let imports = match get_imports_map_from_file(&file_path) {
-                                        Ok(imps) => imps,
-                                        Err(e) => {
-                                            eprintln!("Error {}", e);
-                                            continue;
+                                    let source_file_path_str =
+                                        source_file_path.unwrap().to_string();
+
+                                    let imports = if ignored_dirs
+                                        .iter()
+                                        .any(|r| r.is_match(&source_file_path_str))
+                                    {
+                                        HashMap::new()
```

Comment:
This kind of filtering is sub-optimal-- we still _traverse_ ignored directories, which means we do all the disk I/O necessary to load the extra directory inodes.

You should be filtering just below here in the children.retain() callback, and returning false if the full path matches

(join [children.parent_path](https://docs.rs/jwalk/latest/jwalk/struct.DirEntry.html#structfield.parent_path) and [children.file_name](https://docs.rs/jwalk/latest/jwalk/struct.DirEntry.html#structfield.file_name))

Reply:
Thanks! I think I misunderstood what retain was doing